core/src/test/java/org/apache/calcite/test/ObjectArrayTableTest.java

```diff
@@ -241,6 +244,35 @@ private void withJavaSqlDateTypes(TestWithConnection test) throws SQLException {
     });
 
   }
+
+  /**
+   * Test to confirm java.sql.Timestamp can be casted to Timestamp with different precision.
+   * The cast should be done the same as with TIMESTAMP literal.
+   */
+  @Test public void testCastTimestampToTimestamp() throws SQLException {
+
+    withJavaSqlDateTypes(connection -> {
+
+      Statement statement = connection.createStatement();
+      final String sql = "select %s as EXPECTED, %s as ACTUAL from \"java_sql_date_types\"";
+
+      for (int i = 3; i >= 0; i--) {
+        final String castTo = format(Locale.ROOT, "TIMESTAMP(%d)", i);
```

Comment:
You seem to use `format(Locale.ROOT` extensively. It makes the code extremely obscure. Could you just use `+` operator?

Alternative option is to use Kotlin's string interpolation

Reply:
Changed to simple string concatenation using `+`.